src/orchestrator.ts

```diff
@@ -736,15 +808,13 @@ export class Orchestrator {
     }
 
     /* Returns undefined if not found. */
-    private findTaskScheduled(state: HistoryEvent[], name: string): TaskScheduledEvent | undefined {
-        const returnValue = name
-            ? state.filter((val: HistoryEvent) => {
-                return val.EventType === HistoryEventType.TaskScheduled
+    private findTaskScheduled(state: HistoryEvent[], name: string, startIndex: number = 0): TaskScheduledEvent | undefined {
```

Comment:
Unless I'm mistaken, I don't see `startIndex` being passed explicitly at any point, so I'm unsure of how to use it

Reply:
I think this was a holdover from a previous iteration. I got rid of it in the rebase.